test/integration/clustercidr/ipam_test.go

```diff
@@ -138,6 +139,456 @@ func TestIPAMMultiCIDRRangeAllocatorType(t *testing.T) {
+				makeNode("dualstack-node-2", map[string]string{"ipv4": "true", "ipv6": "true"}),
+				makeNode("dualstack-node-3", map[string]string{"ipv4": "true", "ipv6": "true"}),
+			},
+			expectedPodCIDRs: map[string][]string{
+				"dualstack-node":   {"192.168.0.0/24", "fd00:30:100::/120"},
+				"dualstack-node-2": {"192.168.1.0/24", "fd00:30:100::100/120"},
+				"dualstack-node-3": {"192.168.0.0/24", "fd00:30:100::/120"},
+			},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			if test.clusterCIDR != nil {
+				// Create the test ClusterCIDR
+				if _, err := clientSet.NetworkingV1alpha1().ClusterCIDRs().Create(context.TODO(), test.clusterCIDR, metav1.CreateOptions{}); err != nil {
+					t.Fatal(err)
+				}
+
+				// Wait for the ClusterCIDR to be created
+				if err := wait.PollImmediate(time.Second, 5*time.Second, func() (bool, error) {
+					cc, err := clientSet.NetworkingV1alpha1().ClusterCIDRs().Get(context.TODO(), test.clusterCIDR.Name, metav1.GetOptions{})
+					if err != nil {
+						return false, err
+					}
+					return cc != nil, nil
+				}); err != nil {
+					t.Fatalf("failed while waiting for ClusterCIDR %q to be created: %v", test.clusterCIDR.Name, err)
+				}
+			}
+
+			// Sleep for one second to make sure the controller process the new created ClusterCIDR.
+			time.Sleep(1 * time.Second)
+
+			// Create 1st node and validate that Pod CIDRs are correctly assigned.
+			if _, err := clientSet.CoreV1().Nodes().Create(context.TODO(), test.nodes[0], metav1.CreateOptions{}); err != nil {
+				t.Fatal(err)
+			}
+			if gotPodCIDRs, err := nodePodCIDRs(clientSet, test.nodes[0].Name); err != nil {
+				t.Fatal(err)
+			} else if !reflect.DeepEqual(test.expectedPodCIDRs[test.nodes[0].Name], gotPodCIDRs) {
+				t.Errorf("unexpected result, expected Pod CIDRs %v but got %v", test.expectedPodCIDRs[test.nodes[0].Name], gotPodCIDRs)
+			}
+
+			// Create 2nd node and validate that Pod CIDRs are correctly assigned.
+			if _, err := clientSet.CoreV1().Nodes().Create(context.TODO(), test.nodes[1], metav1.CreateOptions{}); err != nil {
+				t.Fatal(err)
+			}
+			if gotPodCIDRs, err := nodePodCIDRs(clientSet, test.nodes[1].Name); err != nil {
+				t.Fatal(err)
+			} else if !reflect.DeepEqual(test.expectedPodCIDRs[test.nodes[1].Name], gotPodCIDRs) {
+				t.Errorf("unexpected result, expected Pod CIDRs %v but got %v", test.expectedPodCIDRs[test.nodes[1].Name], gotPodCIDRs)
+			}
+
+			// Delete the 1st node, to validate that the PodCIDRs are released.
+			if err := clientSet.CoreV1().Nodes().Delete(context.TODO(), test.nodes[0].Name, metav1.DeleteOptions{}); err != nil {
+				t.Fatal(err)
+			}
+
+			// Create 3rd node, validate that it has Pod CIDRs assigned from the released CIDR.
+			if _, err := clientSet.CoreV1().Nodes().Create(context.TODO(), test.nodes[2], metav1.CreateOptions{}); err != nil {
+				t.Fatal(err)
+			}
+			if gotPodCIDRs, err := nodePodCIDRs(clientSet, test.nodes[2].Name); err != nil {
```

Comment:
the test assumes you always have exactly 3 nodes , the table is not really needed here

Reply:
Done, yeah it turned out to be a single test